beta-src/src/components/map/WDMapController.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from "react";
+import * as d3 from "d3";
+import { Devices } from "../../interfaces";
+import Scale from "../../types/Scale";
+import WDMap from "./WDMap";
+
+const Scales: Scale = {
+  desktop: [0.65, 3],
+  mobileLg: [0.82, 1.6],
+  mobileLgLandscape: [0.3, 1.6],
+  mobile: [0.7, 1.6],
+  mobileLandscape: [0.27, 1.6],
+  tablet: [0.77, 3],
+  tabletLandscape: [0.6, 3],
+};
+
+const getInitialScaleForDevice = (device: keyof Devices): number[] => {
+  if (Object.keys(Scales).includes(device)) {
+    return Scales[device];
+  }
+  return Scales.desktop;
+};
+
+interface WDMapControllerProps {
+  device: keyof Devices;
+  viewportHeight: number;
+  viewportWidth: number;
+}
+
+const WDMapController: React.FC<WDMapControllerProps> = function ({
+  device,
+  viewportHeight,
+  viewportWidth,
+}): React.ReactElement {
+  const svgElement = React.useRef<HTMLElement & SVGElement>(null);
+  const mapOriginalWidth = 4018;
+  const mapOriginalHeight = 2002;
+
+  const initialScaleForDevice = getInitialScaleForDevice(device);
+  const scaleMin = initialScaleForDevice[0];
+  const scaleMax = initialScaleForDevice[1];
+
+  React.useEffect(() => {
+    const fullMap = d3.select(svgElement.current);
```

Comment:
i think we only want to do this when svgElement.current is not null and the ref would be a dependency on this useEffect.